src/data-table.component.html

```diff
@@ -61,7 +61,7 @@
     </ng-container>
   </div>
   <div class="horizontal-scroll">
-    <table mat-table [id]="customId" [dataSource]="dataSource" multiTemplateDataRows cdkDropListGroup matSort>
+    <table mat-table [id]="customId" [dataSource]="dataSource" (contentChanged)="tableContentChanged()" multiTemplateDataRows cdkDropListGroup matSort>
```

Comment:
this could be an `@Input` value and simply just call the proper function in your application.